### PR TITLE
AOM-63: Enable omod files to be uploaded

### PR DESCRIPTION
**Pull request by mkiterian**
## JIRA TICKET NAME:
[AOM-63 Enable omod files to be uploaded](https://issues.openmrs.org/browse/AOM-63)
### SUMMARY:
This PR emanbles omod files(Modules) to be uploaded via the Add-on manager in addition to zip files(OWAs)

### CODE REVIEW

File: app/js/components/manageApps/ManageApps.jsx [hunk 1]
```diff
@@ -143,64 +147,103 @@ export default class ManageApps extends React.Component {
...
+                  toBeInstalledAddonName,
+                  'downgrade',
+                  'downgrading');
+              }
             }
-          }
-        });
-        this.state.addonAlreadyInstalled === false && this.handleUploadRequest();
+          });
+          this.state.addonAlreadyInstalled === false && this.handleUploadRequest();
```
Comment (LWanjiru) [inline app/js/components/manageApps/ManageApps.jsx:207]: Clarification: What does this line of code do exactly?

Comment (mkiterian) [reply app/js/components/manageApps/ManageApps.jsx:207]: `this.state.addonAlreadyInstalled === false` short circuits `this.handleUploadRequest`, in that the latter is only run after the expression is true. In this case if an add-on is already installed then the upload request is not sent. That being said, I did not write this code, what has changed in this line is only the indentation 😄

Comment (LWanjiru) [reply app/js/components/manageApps/ManageApps.jsx:207]: 👍

**Comment by brayoh** [comment]
LGTM.

**Review by tduyile04 — COMMENTED** [review]
Good job 👍

File: app/js/components/manageApps/ManageApps.jsx [hunk 2]
```diff
@@ -143,64 +147,103 @@ export default class ManageApps extends React.Component {
...
+          .then((zip) => {
+            return zip.files['manifest.webapp'] ? zip.files['manifest.webapp'] : false;
+          })
+          .then((file) => {
+            return file ? new_zip.file(file.name).async("string") : false;
+          })
+          .then(result => {
+            resolve(JSON.parse(result));
+          })
+          .catch(e => {
```
Comment (cnagadya) [inline app/js/components/manageApps/ManageApps.jsx:164]: I would recommend you use better variable naming eg error / exception instead of 'e'

**Pull request merged**